internal/wait/pod.go

```diff
@@ -109,3 +109,49 @@ func (wt Waiter) ForPodListAllRunning(ctx context.Context, pods []*corev1.Pod) (
 	wg.Wait()
 	return failed, updated
 }
+
+func (wt Waiter) ForPodListAllDeleted(ctx context.Context, pods []corev1.Pod) ([]*corev1.Pod, []*corev1.Pod) {
+	var lock sync.Mutex
+	var runningPods []*corev1.Pod
+	var deletedPods []*corev1.Pod
+
+	if len(pods) == 0 {
+		klog.Infof("ForPodListAllDeleted called with an empty list of pods. Nothing to do.")
+		return runningPods, deletedPods
+	}
+
+	klog.Infof("Waiting for %d pod(s) to be deleted.", len(pods))
+
+	var wg sync.WaitGroup
+	for _, pod := range pods {
+		wg.Add(1)
+		go func(currentPod *corev1.Pod) {
+			defer wg.Done()
+
+			podKey := client.ObjectKeyFromObject(currentPod).String()
+			klog.Infof("Goroutine started: waiting for pod %s to be deleted.", podKey)
+			err := wt.ForPodDeleted(ctx, currentPod.Namespace, currentPod.Name)
+
+			lock.Lock()
+			if err != nil {
+				klog.Warningf("Failed to confirm deletion for pod %s: %v", podKey, err)
+				runningPods = append(runningPods, currentPod)
+			} else {
+				klog.Infof("Successfully confirmed deletion for pod %s.", podKey)
+				deletedPods = append(deletedPods, currentPod)
+			}
+			lock.Unlock()
```

Comment:
can you clarify why this block is needed beside reporting? I believe this is already checked in the next verification step